components/base/sensorcontrolled/spin.go

```diff
@@ -127,18 +132,15 @@ func getYawInDeg(ctx context.Context, orientation movementsensor.MovementSensor)
 // calcSlowDownAng computes the angle at which the spin should begin to slow down.
 // This helps to prevent overshoot when reaching the goal and reduces the jerk on the robot when the spin is complete.
 func calcSlowDownAng(angleDeg float64) float64 {
-	return math.Min(angleDeg*slowDownAngGain, maxSlowDownAng)
+	return math.Min(math.Abs(angleDeg)*slowDownAngGain, maxSlowDownAng)
```

Comment:
added absolute value here to ensure only the angle error defines the direction of the spin